src/shared/utils/vis.test.ts

```diff
@@ -141,3 +144,53 @@ describe('getMainColumnName', () => {
     ).toEqual('')
   })
 })
+
+describe('getPrecisionTrimmingTableValue - get precision value for geo coordinate calculation', () => {
+  it('calculates a precision value that is used to retrieve id for geo coordinate calculation', () => {
+    const precisionTrimmingTable = getPrecisionTrimmingTableValue()
+
+    expect(precisionTrimmingTable.length - 1).toEqual(HEX_DIGIT_PRECISION)
+    expect(
+      precisionTrimmingTable.filter(num => typeof num === 'bigint').length
+    ).toEqual(HEX_DIGIT_PRECISION + 1)
+  })
+})
+
+describe('getGeoCoordinates - retrieve latitude and longitude values for map geo type', () => {
+  it('returns a latitude and longitude value with key names lat and lon if table with proper columns exists', () => {
+    const table = ({
+      getColumn: () => [0, 1, 2, '123'],
+    } as unknown) as Table
+    const geoCoordinates = getGeoCoordinates(table, 3)
+    expect(geoCoordinates).toEqual(
+      expect.objectContaining({
+        lon: expect.any(Number),
+        lat: expect.any(Number),
```

Comment:
the pattern is identical, and i have tested it with different numbers and the lat lons all come back as you'd expect, though im not sure what the underlying mechanism is with in s2 to get those specific numbers are. If we are testing behavior rather than implementation, id think that making sure the values are numbers is sufficient here. Not sure why we'd need more test cases?

Reply:
>id think that making sure the values are numbers is sufficient here. Not sure why we'd need more test cases?

because this doesn't actually test anything. typescript already ensures any code we write will conform to the interface we specify, and this code is testing that as well. it doesn't assert correctness, it just asserts that the function's interface is correct. it's the coarsest test you can write, and i think it's better to have no test than tests that are this coarse, because again, typescript does this for us already.

>If we are testing behavior rather than implementation

that's exactly what you're testing when you assert that what comes back from the function is what you'd expect given the inputs you pass it? that is the behavior of the function. testing the implementation would be asserting that this function calls the helper function and it returns properly. we don't care about that. we do care about knowing what comes back from this function.

if it's too difficult to explain the masking mechanism it uses to figure out cell ids, just remove this test case.